src/overlays/actors/ovl_Dm_Char08/z_dm_char08.c

```diff
@@ -28,71 +63,1061 @@ const ActorInit Dm_Char08_InitVars = {
+    { &gTurtleSwimAnim, 1.0f, 0.0f, -1.0f, ANIMMODE_LOOP, -24.0f },
+    { &gTurtleFloatAnim, 1.0f, 0.0f, -1.0f, ANIMMODE_LOOP, -24.0f },
+    { &gTurtleSpeak1Anim, 1.0f, 0.0f, -1.0f, ANIMMODE_LOOP, -24.0f },
+    { &gTurtleCoughAnim, 1.0f, 0.0f, -1.0f, ANIMMODE_LOOP, -24.0f },
+    { &gTurtleSpeak2Anim, 1.0f, 0.0f, -1.0f, ANIMMODE_LOOP, -24.0f },
+    { &gTurtleYawnAnim, 1.0f, 0.0f, -1.0f, ANIMMODE_LOOP, -24.0f },
+};
+
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 4000, ICHAIN_STOP),
 };
 
-#endif
+void DmChar08_UpdateEyes(DmChar08* this) {
+    switch (this->eyeMode) {
+        case TURTLE_EYEMODE_0:
+            this->eyeIndex = 0;
+            if (this->blinkTimer > 0) {
+                this->blinkTimer--;
+            } else {
+                this->blinkTimer = 0;
+            }
+            if (this->blinkTimer < 4) {
+                this->eyeIndex = this->blinkTimer;
+            }
+            if (this->blinkTimer == 0) {
+                this->blinkTimer = Rand_S16Offset(30, 30);
+            }
+            break;
+
+        case TURTLE_EYEMODE_1:
+            this->eyeIndex = 4;
+            if (this->blinkTimer > 0) {
+                this->blinkTimer--;
+            } else {
+                this->blinkTimer = 0;
+            }
+            if (this->blinkTimer < 4) {
+                this->eyeIndex = this->blinkTimer + 4;
+            }
+            if (this->blinkTimer == 0) {
+                this->blinkTimer = Rand_S16Offset(30, 30);
+            }
+            break;
+
+        case TURTLE_EYEMODE_2:
+            this->eyeIndex = 2;
+            break;
+
+        case TURTLE_EYEMODE_3:
+            this->eyeIndex = 4;
+            break;
+
+        case TURTLE_EYEMODE_5:
+            this->eyeIndex = 8;
+            break;
+
+        default:
+            break;
+    }
+}
+
+void DmChar08_ChangeAnim(SkelAnime* skelAnime, AnimationInfo* animationInfo, u16 animIndex) {
+    f32 endFrame;
+
+    animationInfo += animIndex;
+    if (animationInfo->frameCount < 0.0f) {
+        endFrame = Animation_GetLastFrame(animationInfo->animation);
+    } else {
+        endFrame = animationInfo->frameCount;
+    }
+    Animation_Change(skelAnime, animationInfo->animation, animationInfo->playSpeed, animationInfo->startFrame, endFrame,
+                     animationInfo->mode, animationInfo->morphFrames);
+}
+
+void DmChar08_Init(Actor* thisx, PlayState* play2) {
+    PlayState* play = play2;
+    DmChar08* this = THIS;
+
+    thisx->targetMode = 5;
+    this->eyeMode = TURTLE_EYEMODE_2;
+    thisx->targetArrowOffset = 120.0f;
+    ActorShape_Init(&thisx->shape, 0.0f, ActorShadow_DrawCircle, 24.0f);
+    SkelAnime_InitFlex(play, &this->skelAnime, &gTurtleSkel, NULL, NULL, NULL, 0);
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
```

Comment:
Are you required to use `&this->dyna.actor` here, or can you use `thisx` like other parts of this function?
```suggestion
    Actor_ProcessInitChain(thisx, sInitChain);
```